File: src/main/java/uk/co/eelpieconsulting/feedlistener/model/TwitterTagSubscription.java
package uk.co.eelpieconsulting.feedlistener.model;

import dev.morphia.annotations.Entity;
import org.apache.commons.codec.digest.DigestUtils;

@Entity("subscriptions")
public class TwitterTagSubscription extends Subscription {

	private String tag;
	
	public TwitterTagSubscription() {
	}

	public TwitterTagSubscription(String tag, String channel, String username) {
		this.setId("twitter-" + DigestUtils.md5Hex("tag" + tag));
		this.setTag(tag);
		this.setChannelId(channel);
		this.setUsername(username);
	}
	
	public String getTag() {
		return tag;
	}
	
	public void setTag(String tag) {
		this.tag = tag;
		this.setName(generateName(tag));
	}

	@Override
	public String toString() {
		return "TwitterTagSubscription [tag=" + tag + "]";
	}
	
	private String generateName(String tag) {
		return "Twitter - " + tag;
	}
	
}

File: src/main/kotlin/uk/co/eelpieconsulting/feedlistener/model/TwitterTagSubscription.kt
package uk.co.eelpieconsulting.feedlistener.model

import dev.morphia.annotations.Entity
import org.apache.commons.codec.digest.DigestUtils

@Entity("subscriptions")
class TwitterTagSubscription(val tag: String, channelId: String, username: String) : Subscription() {

    init {
        id = "twitter-" + DigestUtils.md5Hex("tag$tag")
        name = generateName(tag)
        this.channelId = channelId
        this.username = username
    }

    override fun toString(): String {
        return "TwitterTagSubscription [tag=$tag]"
    }

    private fun generateName(tag: String): String {
        return "Twitter - $tag"
    }

}